tests/test_threat_intelligence.py

```diff
@@ -99,8 +113,731 @@ def test_check_local_ti_files_for_update(
+        # Test CNAME not found
+        mock_db.is_domain_malicious.return_value = (False, False)  # Return (False, False) or ({}, False)
+
+    result = threatintel_obj.is_malicious_cname(
+        "example.com",
+        cname,
+        "uid123",
+        "2023-11-28 12:00:00",
+        "profile_10.0.0.1",
+        "timewindow1",
+    )
+    assert result == expected_result
+
+@pytest.mark.parametrize(
+    "offline_result, online_result, expected_result",
+    [
+        ({"description": "Malicious IP", "source": "test_source"}, None, True),  # Offline hit
+        (None, {"description": "Malicious IP", "source": "test_source"}, True),  # Online hit
+        (None, None, False),  # No hit
+    ],
+)
+def test_is_malicious_ip(
+    threatintel_obj, mocker, offline_result, online_result, expected_result
+):
+    """Test `is_malicious_ip` for checking IP blacklisting."""
+
+    with patch(
+        "modules.threat_intelligence.threat_intelligence.ThreatIntel.search_offline_for_ip",
+        return_value=offline_result,
+    ), patch(
+        "modules.threat_intelligence.threat_intelligence.ThreatIntel.search_online_for_ip",
+        return_value=online_result,
+    ):
+        result = threatintel_obj.is_malicious_ip(
+            "192.168.1.1",
+            "uid123",
+            "10.0.0.1",
+            "2023-11-28 12:00:00",
+            "profile_10.0.0.1",
+            "timewindow1",
+            "srcip",
+        )
+        assert result == expected_result
+
+def test_is_malicious_domain(threatintel_obj, mocker):
+    """
+    Test `is_malicious_domain` for identifying and recording evidence of malicious domains.
+    """
+    mock_search_offline_for_domain = mocker.patch.object(
+        threatintel_obj, "search_offline_for_domain"
+    )
+    mock_set_evidence_malicious_domain = mocker.patch.object(
+        threatintel_obj, "set_evidence_malicious_domain"
+    )
+    # Testcase 1:scenario where the domain is found to be malicious
+    mock_search_offline_for_domain.return_value = (
+        {"description": "Malicious domain"},
+        False,
+    )
+    threatintel_obj.is_malicious_domain(
+        "example.com", "uid123", "2023-11-28 12:00:00", "profile_10.0.0.1", "timewindow1"
+    )
+    mock_set_evidence_malicious_domain.assert_called_once()
+    mock_search_offline_for_domain.reset_mock()
+    mock_set_evidence_malicious_domain.reset_mock()
+    # Testcase 2:scenario where the domain is not found to be malicious
+    mock_search_offline_for_domain.return_value = (None, False)
+    threatintel_obj.is_malicious_domain(
+        "safe.com", "uid123", "2023-11-28 12:00:00", "profile_10.0.0.1", "timewindow1"
+    )
+    mock_set_evidence_malicious_domain.assert_not_called()
+
+@pytest.mark.parametrize(
+    "should_update_return_value, expected_parse_call, expected_db_call",
+    [
+        ("new_hash", True, True),  # File needs update
+        (False, False, False),  # File doesn't need update
+    ],
+)
+def test_update_local_file(
```

Comment:
we also need to test if appropriate handler functions we called, for example, JARM files shoudl result in a call to parse_jarm_file(), JA3 -> parse_ja3_file() , etc